Core/include/Acts/EventData/TrackStateProxy.hpp

```diff
@@ -615,145 +597,101 @@ class TrackStateProxy {
-  }
-
-  template <std::size_t measdim>
-  SubspaceIndices<measdim> subspaceIndices() const {
-    BoundSubspaceIndices boundSubspace = BoundSubspaceIndices();
-    SubspaceIndices<measdim> subspace;
-    std::copy(boundSubspace.begin(), boundSubspace.begin() + measdim,
-              subspace.begin());
-    return subspace;
-  }
-
-  void setBoundSubspaceIndices(BoundSubspaceIndices boundSubspace)
+  /// Set the projector subspace indices
+  /// @param boundSubspace The projector subspace indices to set
+  void setProjectorSubspaceIndices(BoundSubspaceIndices boundSubspace)
     requires(!ReadOnly)
   {
     assert(has<hashString("projector")>());
     component<SerializedSubspaceIndices, hashString("projector")>() =
         serializeSubspaceIndices(boundSubspace);
   }
 
+  /// Set the projector subspace indices
+  /// @param subspace The projector subspace indices to set
   template <std::size_t measdim>
-  void setSubspaceIndices(SubspaceIndices<measdim> subspace)
+  void setProjectorSubspaceIndices(SubspaceIndices<measdim> subspace)
     requires(!ReadOnly && measdim <= eBoundSize)
   {
     assert(has<hashString("projector")>());
     BoundSubspaceIndices boundSubspace{};
     std::copy(subspace.begin(), subspace.end(), boundSubspace.begin());
-    setBoundSubspaceIndices(boundSubspace);
+    setProjectorSubspaceIndices(boundSubspace);
   }
 
+  /// Set the projector subspace indices
+  /// @param subspaceIndices The projector subspace indices to set
   template <std::size_t measdim, typename index_t>
-  void setSubspaceIndices(std::array<index_t, measdim> subspaceIndices)
+  void setProjectorSubspaceIndices(std::span<index_t, measdim> subspaceIndices)
     requires(!ReadOnly && measdim <= eBoundSize)
   {
     assert(has<hashString("projector")>());
     BoundSubspaceIndices boundSubspace{};
     std::transform(subspaceIndices.begin(), subspaceIndices.end(),
                    boundSubspace.begin(),
                    [](index_t i) { return static_cast<std::uint8_t>(i); });
-    setBoundSubspaceIndices(boundSubspace);
+    setProjectorSubspaceIndices(boundSubspace);
+  }
+
+  /// Set the projector subspace indices
+  /// @param subspaceIndices The projector subspace indices to set
+  template <typename index_t>
+  void setProjectorSubspaceIndices(std::span<index_t> subspaceIndices) {
+    assert(has<hashString("projector")>());
+    assert(subspaceIndices.size() <= eBoundSize);
+    BoundSubspaceIndices boundSubspace{};
+    std::transform(subspaceIndices.begin(), subspaceIndices.end(),
+                   boundSubspace.begin(),
+                   [](index_t i) { return static_cast<std::uint8_t>(i); });
+    setProjectorSubspaceIndices(boundSubspace);
+  }
+
+  /// Set the projector subspace indices
+  /// @param subspaceIndices The projector subspace indices to set
+  template <std::size_t measdim, typename index_t>
+  void setProjectorSubspaceIndices(std::array<index_t, measdim> subspaceIndices)
```

Comment:
Is this overload even needed? Can't we make the caller construct the span?

Reply:
Somehow the a `std::array` will not auto-convert to a `std::span` and the caller would need to convert by hand. So I thought it might be easier to have another overload as `std::array` seems like the usual case